lib/modules/ai_bot/commands/search_command.rb

```diff
@@ -154,17 +154,33 @@ def process(**search_args)
       end
 
       @last_num_results = posts.length
+      # this is the general pattern from core
+      # if there are millions of hidden tags it may fail
+      hidden_tags = nil
 
       if posts.blank?
         { args: search_args, rows: [], instruction: "nothing was found, expand your search" }
       else
         format_results(posts, args: search_args) do |post|
-          {
+          category_names = [
+            post.topic.category&.parent_category&.name,
```

Comment:
I don't know how the search command is executed in the background. Will it cause n+1 in category and parent_category attributes too?